model/build.gradle.kts

```diff
@@ -109,27 +111,44 @@ tasks {
     }
 }
 
+/**
+ * Publishing is configured for maven central by default.
+ * Repository can be configured using a command line argument:
+ * ```
+ * gradle publish -DrepoUrl="https://example-repo.com/"
+ * ```
+ */
 publishing {
 
     repositories {
         maven {
             val isSnapshot: Boolean = version.toString().endsWith("SNAPSHOT")
-            val snapshotRepo: URI = URI.create("https://s01.oss.sonatype.org/content/repositories/snapshots/")
-            val releaseRepo: URI = URI.create("https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/")
+            val snapshotRepo2: URI = URI.create("https://s01.oss.sonatype.org/content/repositories/snapshots/")
+            val releaseRepo2: URI = URI.create("https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/")
+            val mavenRepo: URI = if (isSnapshot) snapshotRepo2 else releaseRepo2
+
+            val commandLineRepoUrl = System.getProperty("repoUrl")
 
             name = "OSSRH"
-            url = if (isSnapshot) snapshotRepo else releaseRepo
+            url = if (commandLineRepoUrl != null) URI.create(commandLineRepoUrl) else mavenRepo
```

Comment:
simplified this section: by default `gradle publish` publishes on maven central, and a custom repo can be passed `gradle publish -DrepoUrl="..."`